config/sampler_config.go

```diff
@@ -36,3 +40,38 @@ type TotalThroughputSamplerConfig struct {
 	AddSampleRateKeyToTrace      bool
 	AddSampleRateKeyToTraceField string `validate:"required_with=AddSampleRateKeyToTrace"`
 }
+
+type RulesBasedSamplerCondition struct {
+	Field    string
+	Operator string
+	Value    interface{}
+}
+
+func (r *RulesBasedSamplerCondition) String() string {
+	return fmt.Sprintf("%+v", *r)
+}
+
+type RulesBasedDownstreamSampler struct {
+	DynamicSampler    *DynamicSamplerConfig
+	EMADynamicSampler *EMADynamicSamplerConfig
+}
+
+type RulesBasedSamplerRule struct {
+	Name       string
+	SampleRate int
+	Downstream *RulesBasedDownstreamSampler
```

Comment:
I agree `Downstream` doesn't feel like the right word. Maybe we can use `Sampler`, so we'd have both a `rule.Sampler` and `rule.SampleRate`.